tests/roberta/test_modeling_tf_roberta.py

```diff
@@ -143,21 +143,361 @@ def create_and_check_roberta_model(
+        }
+        result = model(inputs)
+
+        inputs = [input_ids, input_mask]
+        result = model(inputs, token_type_ids=token_type_ids, encoder_hidden_states=encoder_hidden_states)
+
+        # Also check the case where encoder outputs are not passed
+        result = model(input_ids, attention_mask=input_mask, token_type_ids=token_type_ids)
+
+        self.parent.assertEqual(result.last_hidden_state.shape, (self.batch_size, self.seq_length, self.hidden_size))
+
+    def create_and_check_causal_lm_model(
+        self, config, input_ids, token_type_ids, input_mask, sequence_labels, token_labels, choice_labels
+    ):
+        config.is_decoder = True
+
         model = TFRobertaForCausalLM(config=config)
-        result = model([input_ids, input_mask, token_type_ids])
-        self.parent.assertEqual(result.logits.shape, (self.batch_size, self.seq_length, self.vocab_size))
+        inputs = {
+            "input_ids": input_ids,
+            "attention_mask": input_mask,
+            "token_type_ids": token_type_ids,
+        }
+        prediction_scores = model(inputs)["logits"]
+        self.parent.assertListEqual(
+            list(prediction_scores.numpy().shape), [self.batch_size, self.seq_length, self.vocab_size]
+        )
+
+    def create_and_check_causal_lm_model_as_decoder(
+        self,
+        config,
+        input_ids,
+        token_type_ids,
+        input_mask,
+        sequence_labels,
+        token_labels,
+        choice_labels,
+        encoder_hidden_states,
+        encoder_attention_mask,
+    ):
+        config.add_cross_attention = True
+
+        model = TFRobertaForCausalLM(config=config)
+        inputs = {
+            "input_ids": input_ids,
+            "attention_mask": input_mask,
+            "token_type_ids": token_type_ids,
+            "encoder_hidden_states": encoder_hidden_states,
+            "encoder_attention_mask": encoder_attention_mask,
+        }
+        result = model(inputs)
+
+        inputs = [input_ids, input_mask]
+        result = model(inputs, token_type_ids=token_type_ids, encoder_hidden_states=encoder_hidden_states)
+
+        prediction_scores = result["logits"]
+        self.parent.assertListEqual(
+            list(prediction_scores.numpy().shape), [self.batch_size, self.seq_length, self.vocab_size]
+        )
+
+    def create_and_check_causal_lm_model_past(
+        self,
+        config,
+        input_ids,
+        token_type_ids,
+        input_mask,
+        sequence_labels,
+        token_labels,
+        choice_labels,
+    ):
+        config.is_decoder = True
+
+        model = TFRobertaForCausalLM(config=config)
+
+        # special to `RobertaEmbeddings` in `Roberta`:
+        #   - its `padding_idx` and its effect on `position_ids`
+        #     (TFRobertaEmbeddings.create_position_ids_from_input_ids)
+        #   - `1` here is `TFRobertaEmbeddings.padding_idx`
+        input_ids = tf.where(input_ids == 1, 2, input_ids)
```

Comment:
This is required for testing Roberta with the past inputs.

Originally, the `input_ids` is generated with `ids_tensor()`, which might contain `1`, i.e. `TFRobertaEmbeddings.padding_idx`.

Considering a simplified example: `input_ids = [7, 1, 9, 1, 4]`, and let it be the initial `present` inputs.
The `position_ids` computed in `TFRobertaEmbeddings.create_position_ids_from_input_ids` would be:

```
incremental_indices:
    [1, 1, 2, 2, 3]

position_ids:
    [2, 2, 3, 3, 4]
```
Now let's add a next token `5` as the new `present`, and `[7, 1, 9, 1, 4]` as the new `past`.
The position for the next token `5` would be


```
incremental_indices:
    [6] (with `past_key_values_length=5`)

position_ids:
    [7] --> the `next_token` has  `position_id 7` 
```

However, if we pass the concatenated input `[7, 1, 9, 1, 4, 5]` to the model without past `past_key_values_length`,
```
incremental_indices:
    [1, 1, 2, 2, 3, 4]

position_ids:
    [2, 2, 3, 3, 4, 5] --> the `next_token` has  `position_id 5` 
```

**Summary**

This will fail the testing for `past` when `padding_idx` appears in the initial `input_ids`.